lorien/UI/Dialogs/AddKeyDialog.gd

```diff
@@ -0,0 +1,64 @@
+extends WindowDialog
+
+# -------------------------------------------------------------------------------------------------
+export var action_name: String
+export var readable_action_name: String
+
+# -------------------------------------------------------------------------------------------------
+var _pending_bind_event = null
+
+# -------------------------------------------------------------------------------------------------
+func _process(delta) -> void:
+	$VBoxContainer/EventText.text = tr(
+		"KEYBINDING_DIALOG_BIND_ACTION"
+	).format({"action": readable_action_name})
+
+# -------------------------------------------------------------------------------------------------
+func _action_for_event(event: InputEvent):
+	for action in Utils.bindable_actions():
+		if InputMap.action_has_event(action, event):
+			return action
+	return null
+
+# -------------------------------------------------------------------------------------------------
+func _input(event: InputEvent) -> void:
+	if ! visible or $ConfirmRebind.visible:
+		return
+	
+	if event is InputEventKey && event.is_pressed():
+		get_tree().set_input_as_handled()
+		
+		if KEY_MODIFIER_MASK & event.scancode != 0:
```

Comment:
What does that do? a bitwise-and between an integer and a boolen?

Reply:
Ah nevermind, i get it. Mayba add parentheses around `KEY_MODIFIER_MASK & event.scancode`

Reply:
Yes i had to check it myself as well. Looked kinda confusing at first sight :)